### DISCUSSION TITLE
Problem with Import_Export Module

### DISCUSSION THREAD

**Discussion by xroot16x** [apostrophecms/apostrophe#4391] [Q&A]
Hi,
We have installed Apostrophe with the Uploadfs module (blob on azure). We have aslo installed the Import-Export module and for this module we have a problem : the export generate a file in a blob on azure but the parent folder of export files has a probleme with his name:  '\<no name\>'.
Where can we configurate the export folder path for the export module ?

**Comment by BoDonkey** [comment]
Can you provide the rest of the directory structure? Also, Is this a single page, a single piece, several pieces, and do the documents have additional attachments like images?

**Comment by xroot16x** [comment]
Hi, 
Thank you for your answer. 


We have exported a single page with image (this step is working). 


BUT the destination folder is not defined (<no name>) and so the url to get tar.gz file is : 


https://xxxyyyzzz.blob.core.windows.net/apostrophe // exports/a3-boilerplate-bon/plan-export-20240117133247.tar.gz 




The URL contains a double '/ ' because the folder has no name... but the import file open by the web browser at the end of the export not content the double '/ ' and so the web browser can open the file correctly 










----- Mail original -----

De: "Robert Means" ***@***.***> 
À: "apostrophecms/apostrophe" ***@***.***> 
Cc: "xroot16x" ***@***.***>, "Author" ***@***.***> 
Envoyé: Mercredi 17 Janvier 2024 12:03:29 
Objet: Re: [apostrophecms/apostrophe] Problem with Import_Export Module (Discussion #4391) 



Can you provide the rest of the directory structure? Also, Is this a single page, a single piece, several pieces, and do the documents have additional attachments like images?

**Comment by BoDonkey** [comment]
And you aren't having this issue with. any other attachment uploads? I'm just trying to determine if it is an error in the import-export module or in configuration of uploads. There isn't an option in the import-export to change the path. I'm fairly certain it is set by this code
```
const downloadPath = path.join('/exports', filename);
      const downloadUrl = `${self.apos.attachment.uploadfs.getUrl()}${downloadPath}`;
      const copyIn = util.promisify(self.apos.attachment.uploadfs.copyIn);
      try {
        await copyIn(filepath, downloadPath);
      } catch (error) {
        await self.removeExportFile(filepath);
        throw error;
      }
```

**Reply by xroot16x** [reply]
Hi. Do you have a clue for a solution ?

**Comment by boutell** [comment]
Hi xroot16x, can you share your complete uploadfs options object? Without passwords of course.

**Reply by xroot16x** [reply]
Hi. Do you have a clue for a solution ?

**Comment by xroot16x** [comment]
module . exports = { 
	options : { 
		uploadfs : { 
			storage : 'azure' , 
			account : process . env . STORAGE_ACCOUNT , 
			container : 'apostrophe' , 
			key : process . env . STORAGE_ACCOUNT_KEY , 
			// Always required for Azure 
			disabledFileKey : 'xxxxx-xxxxxx-xxxxxx-xxxxxxxxxxxxxxxxx' , 	
			uploadsPath : '/TestUpload/' , 	
			tempPath : '/temp' 
		} 
	} 
}; 

I have tested with this 2 parameters but nothing change : 

uploadsPath : '/TestUpload/' , 
tempPath : '/temp' 

And you aren't having this issue with. any other attachment uploads? => A pparently not. Uploadfs module works perfectly for stocking the attachments of our pages

**Reply by xroot16x** [reply]
HI. Do you have a clue for a solution?

**Reply by xroot16x** [reply]
UP !!

**Comment by BoDonkey** [comment]
We have a ticket in to our engineering team. I **think** it is going to take a change to the uploads module and how it handles storage on Azure. If you have access to our support portal you can submit a ticket describing the problem to be updated directly. Alternatively, we would welcome a PR addressing the issue.
Cheers,
Bob
